include/s3select.h

```diff
@@ -1218,14 +1244,27 @@ void push_like_predicate::builder(s3select* self, const char* a, const char* b)
   base_statement* expr = self->getAction()->exprQ.back();
   self->getAction()->exprQ.pop_back();
 
-  if (!dynamic_cast<variable*>(expr))
-  {
-    throw base_s3select_exception("like expression must be a constant string", base_s3select_exception::s3select_exp_en_t::FATAL);
-  }
-  else if( dynamic_cast<variable*>(expr)->m_var_type != variable::var_t::COL_VALUE)
-  {
-    throw base_s3select_exception("like expression must be a constant string", base_s3select_exception::s3select_exp_en_t::FATAL);
-  }
+  func->push_argument(expr);
+
+  base_statement* main_expr = self->getAction()->exprQ.back();
+  self->getAction()->exprQ.pop_back();
+  func->push_argument(main_expr);
+
+  self->getAction()->exprQ.push_back(func);
+}
+
+void push_like_predicate_dynamic::builder(s3select* self, const char* a, const char* b) const
+{
+  // expr like expr ; both should be string, the second expression could be compiled at the
```

Comment:
per each syntax-rule (escape and no escape)

AST-builder will create the following 
  _fn_like(like-expr , escape-expr)
 upon no escape should set escape-expr to NULL